supervisor.go

```diff
@@ -37,14 +38,28 @@ type Supervisor struct {
 	done       chan struct{}
 }
 
-// NewSupervisor create a Supervisor object with supervisor configuration file
-func NewSupervisor(configFile string) *Supervisor {
-	return &Supervisor{
+type optionFunc func(s *Supervisor)
+
+// WithTags returns an option function to configure a Supervisor
+// with a list of tag values to use when evaluating the configuration.
+func WithTags(tags []string) optionFunc {
+	return func(s *Supervisor) {
+		s.tags = tags
+	}
+}
+
+// NewSupervisor create a Supervisor object with supervisor configuration file, using the specified options.
+func NewSupervisor(configFile string, opts ...optionFunc) *Supervisor {
```

Comment:
similar suggestion to `config.Load` - i'd probably just pass in `tags` directly.

Reply:
Done